### PR TITLE
RDS module initial commit

### LINKED ISSUE #6: RDS Aurora Database
**Objectives:**
- Create a RDS Aurora MySQL (5.7+) database to be used by Wordpress. Aurora can be Serverless. 

**Acceptance Criteria:**
- All resources to be created using Terraform
- RDS Aurora Mysql (version 5.7+)
- Serverless version is recommended due to costs
- No need for aurora replica
- Use a random generated password and upload to SSM Parameter Store for future use
- Use custom VPC from network stack (private subnets)
- Backup just for one day retention is fine
- Encryption enabled
- No need for performance insights, enhanced monitoring and IAM authentication
- Disable deletion protection
- Database need to be accessible from public EC2 running on public subnets
- GitHub can't have any secrets commited.

### PR DESCRIPTION
**Pull request by feernandobraga**
The module is functional and it creates the rds instance.
I've added a TODO list to the readme file with things that are missing as the module depends on other modules to work.
I'm not sure what outputs this module will have to provide to other modules.

### CODE REVIEW

File: terraform/modules/rds-aurora-database/main.tf [hunk 1]
```diff
@@ -0,0 +1,99 @@
+// setting the provider
+provider "aws" {
```
Comment (fabioacp) [inline terraform/modules/rds-aurora-database/main.tf:2]: I think you can remove this. We can set the provider when calling the modules.

File: terraform/modules/rds-aurora-database/main.tf [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  version = "~> 2.0"
+  region  = "ap-southeast-2"
+}
+
+
+// defining custom vpc
+// TODO: utilize the project's main VPC
+resource "aws_vpc" "temp_custom_vpc" {
```
Comment (fabioacp) [inline terraform/modules/rds-aurora-database/main.tf:10]: You can remove this. Let's use variables to represent the subnets.

File: terraform/modules/rds-aurora-database/main.tf [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+resource "aws_subnet" "temp_da_rds_subnet2" {
+  vpc_id     = aws_vpc.temp_custom_vpc.id
+  cidr_block = "192.168.52.0/24"
+  availability_zone = "ap-southeast-2b"
+
+  tags = {
+    Name = "Terraform - temp_da_rds_subnet2"
+  }
+}
```
Comment (fabioacp) [inline terraform/modules/rds-aurora-database/main.tf:37]: you can remove the "aws_subnet" resources. Let's use variables for the subnets on db_subnet_group.

File: terraform/modules/rds-aurora-database/main.tf [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+//defining custom subnet for the database
+resource "aws_db_subnet_group" "da_db_subnet_group" {
```
Comment (fabioacp) [inline terraform/modules/rds-aurora-database/main.tf:40]: I'd recommend using variables.

```
resource "aws_db_subnet_group" "da_db_subnet_group" {
  name             = "${var.name}-db-subnet-group"
  subnet_ids       = var.subnets

  tags = merge(
      var.tags,
      {
        "Name"    = "${var.name}-db-subnet-group"
      }
    )
}
```

File: terraform/modules/rds-aurora-database/main.tf [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  tags = {
+    "Name" = "Terraform security group"
+  }
+
+}
+
+
+// defining aurora
+// TODO: implement random passwords with SSM
+resource "aws_rds_cluster" "da-aurora-cluster" {
```
Comment (fabioacp) [inline terraform/modules/rds-aurora-database/main.tf:80]: I'd recommend using variables.

Example:

```
resource "aws_rds_cluster" "da-aurora-cluster" {
  cluster_identifier                  = var.name
  engine                              = var.engine
  engine_mode                         = var.engine_mode
  engine_version                      = var.engine_version
  database_name                       = var.database_name
  master_username                     = var.username
  master_password                     = var.master_password
  backup_retention_period             = var.backup_retention_period
  vpc_security_group_ids              = compact(concat(aws_security_group.this.*.id, var.vpc_security_group_ids))
  db_subnet_group_name                = var.db_subnet_group_name
  skip_final_snapshot                 = var.skip_final_snapshot

  dynamic "scaling_configuration" {
    for_each = length(keys(var.scaling_configuration)) == 0 ? [] : [
    var.scaling_configuration]

    content {
      auto_pause               = lookup(scaling_configuration.value, "auto_pause", null)
      max_capacity             = lookup(scaling_configuration.value, "max_capacity", null)
      min_capacity             = lookup(scaling_configuration.value, "min_capacity", null)
      seconds_until_auto_pause = lookup(scaling_configuration.value, "seconds_until_auto_pause", null)
      timeout_action           = lookup(scaling_configuration.value, "timeout_action", null)
    }
  }
}
```

File: terraform/modules/rds-aurora-database/main.tf [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  vpc_security_group_ids  = [aws_security_group.da_rds_security_group.id]
+  db_subnet_group_name    = aws_db_subnet_group.da_db_subnet_group.name
+  skip_final_snapshot     = true
+
+  // determining the min and max capacity for auto scalling 
+  scaling_configuration {
+    min_capacity = 1
+    max_capacity = 2
+  }
+}
```
Comment (fabioacp) [inline terraform/modules/rds-aurora-database/main.tf:99]: I think it's missing "aws_rds_cluster_instance".

**Review by fabioacp — CHANGES_REQUESTED** [review]
Hi @feernandobraga, thanks for the PR. I just put comments on it. Let me know if you have any questions.

File: terraform/modules/rds-aurora-database/main.tf (hunk 1 above)
Comment (feernandobraga) [reply terraform/modules/rds-aurora-database/main.tf:2]: of course fabio, I just put it there so it could be 'runnable'

Comment (alanlima) [reply terraform/modules/rds-aurora-database/main.tf:2]: @feernandobraga, I usually create a folder in `terraform/examples/my-testing-module` what will be the entry point to call the module that I am working on.

File: terraform/modules/rds-aurora-database/main.tf (hunk 4 above)
Comment (alanlima) [reply terraform/modules/rds-aurora-database/main.tf:40]: @zarajoy created the variable `project-name` in her VPC module. I like the variable name and could be used here.
Ideally, when we start to put everything together, we could have multiple environments (prod, test). We would change the project-name to `devops-wordpress-prod` or `devops-wordpress-test` (example names) and it should not have any conflict.

**Review by alanlima — APPROVED** [review]
Looks good, @feernandobraga might do some extra work this weekend.
Squash/merge this PR and you can create another PR with the new changes.

**Review by fabioacp — APPROVED** [review]

**Pull request merged**